lib/MirrorCache/Datamodule.pm

```diff
@@ -262,6 +263,12 @@ sub redirect($self, $url) {
         $xtra = ".metalink";
     } elsif ($self->_original_path =~ m/.*\.mirrorlist$/) {
         $xtra = ".mirrorlist";
+    } elsif ($self->_original_path =~ m/.*\.torrent$/) {
```

Comment:
why not reduce duplication with
```perl
if ($self->_original_path =~ m/(\.metalink|\.mirrorlist|\.torrent|\.magnet|\.btih)$/) {
    $xtra = $1;
}
```

also the leading `.*` can be omitted and might even be faster to match.